int8[] data
int8 index
